src/Core/SharedKernels/Enumerator/Enumerator/Implementation/Services/AnswerToStringConverter.cs

```diff
@@ -14,68 +14,99 @@ public class AnswerToStringConverter : IAnswerToStringConverter
     {
         public string Convert(object answer, Guid questionId, IQuestionnaire questionnaire)
         {
-            if(!questionnaire.IsPrefilled(questionId)) throw new NotSupportedException("Only identifying question can be converted to string");
-
-            Func<decimal, string> getCategoricalOptionText = null;
+            if (!questionnaire.IsPrefilled(questionId)) 
+                throw new NotSupportedException("Only identifying question can be converted to string");
 
             var questionType = questionnaire.GetQuestionType(questionId);
 
-            if (answer != null)
+            switch (questionType)

```

Comment:
There is a concern that once we've dropped support for string here the data is passing with AnsweredQuestionSynchronizationDto that has property object for values. If it was serialized and persisted in prior times with the string values we'll have an error here.

Reply:
return support